static/js/publisher/form/media.js

```diff
@@ -0,0 +1,271 @@
+
+        blankMedia.push(
+          <div
+            className={classes.join(" ")}
+            key={`blank-${i}`}
+            onClick={this.addImage}
+            onKeyDown={this.keyboardEvent}
+            onFocus={this.focused.bind(this, "new")}
+            onBlur={this.blurred.bind(this, "new")}
+            tabIndex="0"
+            ref={item => (this.blankItem = item)}
+          >
+            <span role="button" className="u-align-text--center">
+              <i className="p-icon--plus" />
+              <br />
+              Add image
+            </span>
+          </div>
+        );
+      } else {
+        blankMedia.push(
+          <div className={classes.join(" ")} key={`blank-${i}`} />
+        );
+      }
+    }
+
+    return blankMedia;
+  }
+
+  renderDeleteState(toDelete) {
+    return (
+      <span className="p-tooltip p-tooltip--btm-center" key="toDelete">
+        {toDelete.length} images to delete.&nbsp;
+        <span className="p-tooltip__message">
+          {toDelete.map(item => (
+            <img
+              className="p-listing-images__tooltip-image"
+              src={item.url}
+              key={`delete-${item.url}`}
+            />
+          ))}
+        </span>
+      </span>
+    );
+  }
+
+  renderUploadState(toUpload) {
+    return (
+      <span className="p-tooltip p-tooltip--btm-center" key="toUpload">
+        {toUpload.length} images to upload.&nbsp;
+        <span className="p-tooltip__message">
+          {toUpload.map(item => (
+            <img
+              className="p-listing-images__tooltip-image"
+              src={item.url}
+              key={`delete-${item.url}`}
+            />
+          ))}
+        </span>
+      </span>
+    );
+  }
+
+  renderStatus() {
+    const toDelete = this.state.mediaData.filter(
+      item => item.status === "delete"
+    );
+    const toUpload = this.state.mediaData.filter(item => item.status === "new");
+
+    const content = [];
+
+    if (toDelete.length > 0) {
+      content.push(this.renderDeleteState(toDelete));
+    }
+    if (toUpload.length > 0) {
+      content.push(this.renderUploadState(toUpload));
+    }
+    return <p>{content}</p>;
+  }
+
+  render() {
+    const mediaList = this.state.mediaData.filter(
+      item => item.status !== "delete"
+    );
+
+    let blankMedia = 0;
+
+    if (mediaList.length < this.props.mediaLimit) {
+      blankMedia = this.props.mediaLimit - mediaList.length;
+    }
+
+    return (
+      <Fragment>
+        {this.renderOverLimit()}
+        <div
+          className="p-listing-images p-fluid-grid"
+          ref={item => (this.holder = item)}
+        >
+          {mediaList.map((item, i) => (
+            <MediaItem
+              key={item.url}
+              url={item.url}
+              type={item.type}
+              status={item.status}
+              remove={this.markForDeletion}
+              overflow={i > 4}
+              focused={this.focused}
+              blurred={this.blurred}
```

Comment:
I find these short function names (`focused`, `blurred`, `remove`) a bit confusing, especially when mixed with other "value" props like `focus`. It's not clear from their name that they are functions/callbacks and it makes the code bit harder to read I think. Naming like `focusHandler` or `onItemFocus` or similar would make it clearer that it's a function and that it's meant to be handling some event.

This is both about naming class functions (`this.focused`) and props of component.